### PR TITLE
Release v1.0.5

### PR DESCRIPTION
**Pull request by prashantv**
Merges #258, #259, #264, #269, #270, #271, #272, #268, #277, #278, #275, #279, #281, #283, #284, #285, #286, #290, #287, #274, #291, #276, #292, #295, #294, #297, #293, #296, #298, #299, #302, #303, #304, #305, #306, #307, #308, #309, #310, #311, #313 and #314 from dev into master for a 1.0.5 release.

### CODE REVIEW

**Comment by breerly** [comment]
I approve of this branching strategy :+1:

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
 =========
 
+# 1.0.5
+
+* Use `context.Context` storage for headers so `thrift.Context` and
+  `tchannel.ContextWithHeaders` can be passed to functions that use
+  `context.Context`, and have them retain headers.
+* `thrift.Server` allows a custom factory to be used for `thrift.Context`
+  creation based on the underlying `context.Context` and headers map.
+* Store goroutine stack traces on channel creation that can be accessed
```
Comment (breerly) [inline CHANGELOG.md]: This seems like a feature, not a bugfix.

**Comment by breerly** [comment]
It looks like most of the things in this release are features, not bug fixes. If you're following semver, then you'll want to increment the minor. not the patch. If there are backwards breaking changes, then you'll want to increment the major.

File: CHANGELOG.md (hunk 1 above)
Comment (prashantv) [inline CHANGELOG.md]: It's kind of a feature, but it's also invisible to the user. It's mostly for introspection, not actually something the user can use directly.

Comment (breerly) [inline CHANGELOG.md]: If it's new functionality (whether exposed to the user or not), then it's not a bugfix, and therefor not a patch.

**Comment by abhinav** [comment]
:+1:

**Pull request merged**